### PR TITLE
Move license settings 

### PR DESCRIPTION
**Pull request by jakelandis**
Continued refactoring of the license service. This commit follows https://github.com/elastic/elasticsearch/pull/94261 
which moved the LicenseService -> ClusterStateLicenseService and re-introduced 
LicenseService as interface. That change did not move the settings to keep the diff small. 

This change moves those settings to a new class LicenseSettings.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/license/LicenseService.java [hunk 1]
```diff
@@ -7,67 +7,79 @@
...
-import java.util.List;
-import java.util.Set;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
+/**
+ * Interface to read the current license. Consumers should generally not need to read the license directly and should instead
+ * prefer {@link XPackLicenseState} to check if a feature is allowed by the license. This interface is not intended to be implemented
+ * by alternative implementations and exists for internal use only.
+ */
+public interface LicenseService extends LifecycleComponent {
```
Comment (jakelandis) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/license/LicenseService.java:21]: Added the contract and javadoc in this PR to help separate the discussion of the contract vs. implementation. However, in practice this contract is really just derived from the implementation with the following goals :
* Attempt to keep the abstraction as leak free as possible .... the Metadata in the contract is a bit of a leak but kinda unavoidable
* Attempt to seperate the mutable vs. non-mutable interfaces. We inject these implementations and almost always only need the immutable view.

Comment (jakelandis) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/license/LicenseService.java]: decided to defer introduction of contract so this has been removed.

**Review by ywangd — APPROVED** [review]
LGTM

It feels that the settings and constants could just be part of `ClusterStateLicenseService` since they are only relevant to this implementation and are not needed for the other (future) implementation. But I don't feel strong either way.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/license/LicenseSettings.java [hunk 2]
```diff
@@ -26,17 +29,17 @@ public interface LicenseService {
...
         License.LicenseType::parse,
-        LicenseService::validateUploadTypesSetting,
+        LicenseSettings::validateUploadTypesSetting,
         Setting.Property.NodeScope
     );
     // pkg private for tests
-    TimeValue NON_BASIC_SELF_GENERATED_LICENSE_DURATION = TimeValue.timeValueHours(30 * 24);
-    Set<License.LicenseType> VALID_TRIAL_TYPES = Set.of(
+    public static final TimeValue NON_BASIC_SELF_GENERATED_LICENSE_DURATION = TimeValue.timeValueHours(30 * 24);
+    public static final Set<License.LicenseType> VALID_TRIAL_TYPES = Set.of(
```
[the remark this answers is not in the document]
Comment (jakelandis) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/license/LicenseSettings.java]: doh... I missed that Intellij defaults to increasing scope when using it's refactoring tools. fixed in [`42b1e95` (#94566)](https://github.com/elastic/elasticsearch/pull/94566/commits/42b1e951c0d2db19434048a0946ff47ee8c20200)

File: x-pack/plugin/core/src/main/java/org/elasticsearch/license/LicenseSettings.java [hunk 3]
```diff
@@ -45,19 +48,20 @@ public interface LicenseService {
...
+    public static final TimeValue LICENSE_EXPIRATION_WARNING_PERIOD = TimeValue.timeValueDays(7);
+    public static final long BASIC_SELF_GENERATED_LICENSE_EXPIRATION_MILLIS =
+        XPackInfoResponse.BASIC_SELF_GENERATED_LICENSE_EXPIRATION_MILLIS;
     /**
      * Max number of nodes licensed by generated trial license
      */
-    int SELF_GENERATED_LICENSE_MAX_NODES = 1000;
-    int SELF_GENERATED_LICENSE_MAX_RESOURCE_UNITS = SELF_GENERATED_LICENSE_MAX_NODES;
+    public static final int SELF_GENERATED_LICENSE_MAX_NODES = 1000;
+    public static final int SELF_GENERATED_LICENSE_MAX_RESOURCE_UNITS = SELF_GENERATED_LICENSE_MAX_NODES;
```
[the remark this answers is not in the document]
Comment (jakelandis) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/license/LicenseSettings.java]: thanks for catching this... fixed in [`42b1e95` (#94566)](https://github.com/elastic/elasticsearch/pull/94566/commits/42b1e951c0d2db19434048a0946ff47ee8c20200)

**Pull request merged**